Uploaded files can be open in a browser

malyMiso opened the pull request:
#### Description, the reason for the PR
Uploaded files can be configured to being streamed to a browser instead of downloading. 

Deployment update: https://github.com/shopsys/deployment/pull/80

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) <!-- Do not forget to update UPGRADE.md -->
- [x] New feature <!-- Do not forget to update docs -->
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)



























































----
:globe_with_meridians: Live Preview:
  - https://mt-view-uploaded-file.odin.shopsys.cloud
  - https://cz.mt-view-uploaded-file.odin.shopsys.cloud
  - https://mt-view-uploaded-file.odin.shopsys.cloud/sk
<!-- SLACK_TIMESTAMP: 1779104161.668439 -->

<!-- Replace -->
----
:globe_with_meridians: Live Preview:
  - https://mt-view-uploaded-file.odin.shopsys.cloud
  - https://cz.mt-view-uploaded-file.odin.shopsys.cloud
  - https://mt-view-uploaded-file.odin.shopsys.cloud/sk
<!-- Replace -->

malyMiso committed:
viewable files can now be displayed directly in browser instead of directly downloading them

vitek-rostislav commented:
I do not think we should close this, let's rather process it :slightly_smiling_face:

TomasLudvik committed:
exposed separate view URL for uploaded files viewable in browser

Instead of silently changing getUploadedFileUrl() to return view URLs
for viewable files, this exposes both URLs explicitly. Callers can now
use getUploadedFileViewUrl() and isUploadedFileViewableInBrowser() to
offer users a choice between viewing and downloading.

grossmannmartin commented on packages/framework/src/Controller/Front/UploadedFileController.php:
```diff
@@ -21,23 +24,58 @@ public function __construct(
+            $uploadedFile = $this->getByIdSlugAndExtension($uploadedFileId, $uploadedFilename);
+
+            if (!$this->uploadedFileFacade->isUploadedFileViewableInBrowser($uploadedFile)) {
+                throw new NotFoundRedirectToStorefrontException(
```
can't we just fallback to download url?

TomasLudvik committed:
storefront product detail files section now shows view and download links

TomasLudvik committed:
added upgrade notes

TomasLudvik committed:
dumped translations

TomasLudvik committed:
file size is now persisted in database and displayed in administration

ShopsysBot committed:
regenerate all screenshots

grossmannmartin commented on packages/administration/translations/messages.en.po:
```diff
@@ -2149,6 +2149,9 @@ msgstr ""
 msgstr ""
 
+msgid "View"
+msgstr "View"
```
English translations should not be filled if are the same

grossmannmartin commented on packages/framework/build.xml:
```diff
@@ -1003,6 +1005,14 @@
+    <target name="recalculate-file-sizes" description="Populates file size for uploaded files and images that do not have it stored.">
+        <exec executable="${path.php.executable}" passthru="true" checkreturn="true">
+            <arg value="${path.bin-console}"/>
+            <arg value="shopsys:recalculate-file-sizes"/>
```
This command will be run after each deploy, but it's needed just once.

@pk16011990, please verify if it's ok by you. Otherwise, we could add some list of executed tasks (similar to database migration table)

pk16011990 replied:
Good point! @TomasLudvik, if it is not too complicated, pls do it.

grossmannmartin commented on packages/framework/src/Command/RecalculateFileSizesCommand.php:
If we want to improve performance even further, we can bypass entity hydration completely and return only the IDs of files without filesize set.
That lower the memory usage and we won't need to clear entity manager

TomasLudvik committed:
added skill for creating Shopsys enums

TomasLudvik committed:
added post deploy tasks that enable running one or every time task after deploy with maintenance page turned off

grossmannmartin commented on packages/framework/src/Command/RunPostDeployTasksCommand.php:
```diff
@@ -0,0 +1,36 @@
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+#[AsCommand(
+    name: 'shopsys:post-deploy:run-tasks',
```
We already have `deploy:` commands, where I think this would fit better.

grossmannmartin commented on packages/framework/src/Component/PostDeploy/Task/PostDeployTaskConfigLoader.php:
```diff
@@ -0,0 +1,127 @@
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Yaml\Yaml;
+
+class PostDeployTaskConfigLoader
```
I think we kinda re-invent the wheel here. 

What about putting post-deploy tasks under shopsys_framework semantic config?
I prototyped it in https://github.com/shopsys/shopsys/commit/83aa91f4443d9ed3a38262ed1a0bc0cd0835da31, and I like that it follows the already established Symfony's bundle configuration way and also we don't need to use custom loader at all.

grossmannmartin commented on packages/framework/src/DependencyInjection/Configuration.php:
```diff
@@ -27,6 +31,48 @@ public function getConfigTreeBuilder(): TreeBuilder
+                                ->children()
+                                    ->enumNode('run')
+                                        ->isRequired()
+                                        ->values((new PostDeployTaskRunEnum())->getAllCases())
```
Nitpick:
With `new` here, the enum may be a true enum as it's not extendable anyway. 

Or it should be `final` to make the intent clear.

I'm absolutely ok with a non-extendable task run list, as I don't think it will be necessary to modify this list.
